include/NeoFOAM/core/Error.hpp

```diff
@@ -2,29 +2,214 @@
+                << "\nLine: " << std::source_location::current().line() << "\n"                    \
+                << cpptrace::generate_trace().to_string() << "\n"
+#else
+/**
+ * @def NF_ERROR_MESSAGE
+ * @brief Macro for generating an error message without debug information.
+ *
+ * This macro generates an error message with the specified message, current
+ * file name, and current line number.
+ *
+ * @param message The error message to be included in the generated message.
+ * @return std::string The generated error message.
+ */
+#define NF_ERROR_MESSAGE(message)                                                                  \
+    "Error: " << message << "\nFile: " << std::source_location::current().file_name()              \
+              << "\nLine: " << std::source_location::current().line() << "\n"
+#endif
+
+/**
+ * @def NF_ERROR_EXIT
+ * @brief Macro for printing an error message and aborting the program.
+ *
+ * This macro prints the specified error message to the standard error stream,
+ * including the current file name and current line number, and then calls
+ * std::abort() to terminate the program.
+ *
+ * @param message The error message to be printed.
+ */
+#define NF_ERROR_EXIT(message)                                                                     \
+    do                                                                                             \
+    {                                                                                              \
+        std::cerr << NF_ERROR_MESSAGE(message);                                                    \
+        std::abort();                                                                              \
+    }                                                                                              \
+    while (false)
+
+/**
+ * @def NF_THROW
+ * @brief Macro for throwing a NeoFOAMException with an error message.
+ *
+ * This macro constructs a std::stringstream to concatenate the specified error
+ * message with the current file name and current line number, and then throws a
+ * NeoFOAMException with the resulting string as the error message.
+ *
+ * @param message The error message to be included in the exception.
+ */
+#define NF_THROW(message)                                                                          \
+    do                                                                                             \
+    {                                                                                              \
+        std::stringstream ss;                                                                      \
+        ss << NF_ERROR_MESSAGE(message);                                                           \
+        throw NeoFOAM::NeoFOAMException(ss.str());                                                 \
+    }                                                                                              \
+    while (false)
```

Comment:
I think this can just be
```suggestion
    throw NeoFOAM::NeoFOAMException(message)
```
I would assume that message is string-like anyway, so the additional stringstream is unnecessary.

Reply:
I think it's usually helpful to add the same context as `NF_ERROR_MESSAGE` (without the stacktrace maybe). At least the line number + calling function are interesting.